contracts/utils/Strings.sol

```diff
@@ -13,26 +13,55 @@ library Strings {
     /**
      * @dev Converts a `uint256` to its ASCII `string` decimal representation.
      */
-    function toString(uint256 value) internal pure returns (string memory) {
-        // Inspired by OraclizeAPI's implementation - MIT licence
-        // https://github.com/oraclize/ethereum-api/blob/b42146b063c7d6ee1358846c198246239e9360e8/oraclizeAPI_0.4.25.sol
+    function toString(uint256 value) internal pure returns (string memory result) {
+        unchecked {
+            if (value < 10) {
+                return string(abi.encodePacked(uint8(value + 48)));
+            }
```

Comment:
We can't remove it entirely, we still need to handle the special case of `0`. That change decreases gas for the case of `0` from 170 to 131, but for other single-digit numbers increases from 170 to 525.

Why do you want to remove it, is it to clean up the logic or for a different reason? If it's only to clean up, IMO the gas savings justify the tiny additional complexity, especially since single-digit conversions seem to be a common use case, but that's just that: an opinion.

Reply:
we can, see my commit

Reply:
Which commit?

Reply:
This is the commit: https://github.com/CodeSandwich/openzeppelin-contracts/commit/8597e3dcc34e3ae5f48a0ad8a19bf3d0f504703c

The problem is that it creates bad output, in all cases except `0` it creates a string with length 1 too big. Plus now **all** single-digit numbers have cost increased 3-fold to >500, not only `1` to `9`.

Reply:
Ok, I see that it'll work just fine, the change in powers of 10 counter the initial length of 1, very clever! But the gas usage for single-digit numbers is still much higher. What are the advantages?

Reply:
Does removal of the special case reduce a lot of bytecode? Lower gas usage is always a good thing, because we don't know how this code will be used. Otherwise this entire PR is pointless, it complicates the logic in exchange for lower gas. IMO we can keep all your changes including `length = 1`, but still restore the single-digit shortcut.